contracts/staking/PoSValidatorManager.sol

```diff
@@ -241,7 +246,16 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         });
     }
 
-    function initializeEndDelegation(bytes32 validationID) external {
+    function initializeEndDelegation(
+        bytes32 validationID,
+        bool includeUptimeProof,
+        uint32 messageIndex
+    ) external {
+        uint64 uptime;
+        if (includeUptimeProof) {
+            uptime = _updateUptime(validationID, messageIndex);
```

Comment:
We should read from state instead of using a return from `_updateUptime`. That way if the validator's uptime has been updated recently, anyone ending their delegation won't have to submit another uptime proof, provided that the latest reported uptime up until now hits the 80% threshold.